### ISSUE TITLE
Menu item icons with multi-word material names

### ISSUE THREAD

**Issue by piotrskibapl** [MrTigreroux/TigerReports#47]
Hi,

I have a problem with changing the punishment menu icon in the config. When it comes to single-word material names (like stone, TNT, etc.), it works fine. However, it looks like I can't use any material that has more than one word in its name (like spruce_sign, birch_sapling, etc.).

I'm talking about the "Config.Punishments.PunishmentX.Item" config value. Is there any way to use these multi-word materials as menu icons?

**Comment by MrTigreroux** [comment]
Hello,
Are you sure you put correctly "Material-" at the beggining of the item, like "Material-Spruce_sign" for example ?
Does it work with the default _config.yml_ file ? There is for example the chainmail_leggings for Speed default reason that should work ([link](https://github.com/MrTigreroux/TigerReports/blob/1d95865a662b32f56f6655d4799ded9469aad1dc/TigerReports/src/config.yml#L112)).
Did you encode your _config.yml_ file in UTF-8  ?
Can you copy/paste me your DefaultReasons section, and give me your Spigot version (of your server), the version of TigerReports you use and a screenshot of the in game result please ?

**Comment by piotrskibapl** [comment]
Hi,
I tried changing these menu icons on the default plugin config to check whether it's not related to some other options or outdated config.
My config.yml file is encoded in UTF-8, and I'm always using the "Material-" prefix for items.

After some investigation:
 - it looks like this problem doesn't apply to all items - some multi-world material names work fine (like these default in the config - _chainmail_leggings_, etc.)
 - in the default config the last punishment menu item icon should be a pumpkin; however, it appears as a carved pumpkin in the game
 - in the default config (DefaultReasons section) there's a _Material-Sign_ item - when I change that to _Material-oak_sign_, it doesn't work (the icon is set to a paper in game)

I'm also attaching a screenshot with changed _Material-Stick_ to _Material-oak_sign_ under _Reason4_ in _DefaultReasons_:
![oak_sign](https://user-images.githubusercontent.com/33459586/103605107-483f1a00-4f13-11eb-8a4b-152321b32967.png)

For testing purposes, I also created a simple _Punishments_ section with 5 random items that aren't parsed correctly by the plugin and appear as a paper in-game:
```
  Punishments:
    Enabled: yes
    # Open the punishments menu or execute the command below when processing a report as True / Ouvrir le menu des sanctions ou exécuter la commande ci-dessous lors du traitement d'un signalement comme Vrai
    
    PunishmentsCommand: none
    # Replace the punishments menu with a command executed by the staff who process the report / Remplacer le menu des sanctions par une commande exécutée par le membre de l'équipe qui traite le signalement
    # Note that the _Reporter_ placeholder will be replaced by the first reporter, not all to avoid a spam of commands / A noter que le placeholder _Reporter_ sera remplacé par le premier signaleur, et non tous pour éviter d'exécuter plusieurs commandes
    
    DefaultReasons: no
    # Use defaults reasons as punishments / Utiliser les raisons par défaut comme des sanctions
    # "PunishCommands:" property must be added below each default reason / La propriété "PunishCommands:" doit être ajoutée en-dessous de chaque raison par défaut
    
    Punishment1:
      Name: "test 1 - name"
      Lore: "test 1 - lore"
      Item: Material-grass_block
      PunishCommands:
      - "test 1"
    Punishment2:
      Name: "test 2 - name"
      Lore: "test 2 - lore"
      Item: Material-sticky_piston
      PunishCommands:
      - "test 2"
    Punishment3:
      Name: "test 3 - name"
      Lore: "test 3 - lore"
      Item: Material-dark_oak_boat
      PunishCommands:
      - "test 3"
    Punishment4:
      Name: "test 4 - name"
      Lore: "test 4 - lore"
      Item: Material-redstone_lamp
      PunishCommands:
      - "test 4"
    Punishment5:
      Name: "test 5 - name"
      Lore: "test 5 - lore"
      Item: Material-polished_blackstone_button
      PunishCommands:
      - "test 5"
 ```

Please let me know if I can help you any further.

**Comment by MrTigreroux** [comment]
Could you run `/icanhasbukkit` command and give me a screenshot of the result in console or ingame chat please ? If it doesn't work, could you check your Spigot version and give me it please ?

**Comment by piotrskibapl** [comment]
Sure, `/icanhasbukkit` output:
![icanhas](https://user-images.githubusercontent.com/33459586/103699044-a23af080-4fa2-11eb-899b-1e0dfdabc645.png)

TigerReports version: 4.11.8

**Comment by MrTigreroux** [comment]
Could you try with a Spigot and not Paper version please ? Maybe try to use an older version of Paper then.
But I read again the code and if it works with some materials, there is no reason that it doesn't work with all. I guess the problem comes from the "snapshot" which is an incomplete versio, that's why you should try a 1.16.1 Spigot and then Paper version for example.

**Comment by piotrskibapl** [comment]
@MrTigreroux I'm sorry for my late reply, but I did not have too much time to check this last weeks.

I tried running the plugin on Spigot `1.16.1`, `1.16.4` and `1.16.5` versions from https://getbukkit.org/download/spigot - it doesn't also work on these builds - sometimes I get paper as a menu item instead of the item specified in the config.

**Comment by PixelDev0** [comment]
Could the issue possibly be with there being no capital after the `-` in `Material-oak_sign`?

**Comment by MrTigreroux** [comment]
No the name is put in correct format, regardless of the capital letters used. It seems that the problem comes from the Spigot API I use. I'm using Spigot 1.9.4 and it seems that it forces Spigot to use 1.8 material names and not the ones of 1.13+. It therefore prevent 1.13+ Spigot servers to use new blocks/items and I will try to fix this keeping the support of  1.8 Spigot servers. But I miss of time for now.

**Comment by MrTigreroux** [comment]
The problem should be fixed in new [version 4.12](https://www.spigotmc.org/resources/tigerreports.25773/update?update=387622). Thank you for reporting it.
